# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1451290011 -3600
#      Mon Dec 28 09:06:51 2015 +0100
# Node ID 31a12af0aaafd649acbc523d8dbbeda6b07f97aa
# Parent  73e92fca3c781603e21afdb4966eb0eebdf2af97
color of editor margins adapted to currently chosen style (bug #46706)

* file-editor-tab.cc (update_lexer): set the margin colors depending on the
  background color of the current style

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -612,19 +612,35 @@ file_editor_tab::update_lexer ()
             }
           _lexer_apis->prepare ();  // prepare apis info
         }
     }
 
   lexer->readSettings (*settings);
 
   _edit_area->setLexer (lexer);
+
   _edit_area->setCaretForegroundColor (lexer->color (0));
   _edit_area->setIndentationGuidesForegroundColor (lexer->color (0));
 
+  // set margin colors depending on selected background color of the lexer
+  QColor bg = lexer->paper (0), fg;
+  int h, s, v, factor=-1;
+  bg.getHsv (&h,&s,&v);
+
+  if (v < 96)
+    factor = 2;
+
+  bg.setHsv (h,s/2,v + factor*40);
+  fg.setHsv (h,s/2,v + (factor+1)*40);
+
+  _edit_area->setMarginsForegroundColor (lexer->color (0));
+  _edit_area->setMarginsBackgroundColor (bg);
+  _edit_area->setFoldMarginColors (bg,fg);
+
   // fix line number width with respect to the font size of the lexer
   if (settings->value ("editor/showLineNumbers", true).toBool ())
     auto_margin_width ();
   else
     _edit_area->setMarginWidth (2,0);
 
 }
 
